nautobot_golden_config/views.py

```diff
@@ -393,8 +393,11 @@ def diff_structured_data(backup_data, intended_data):
             if request.GET.get("format") in ["json", "yaml"]:
                 structure_format = request.GET.get("format")
 
-            settings = get_device_to_settings_map(queryset=Device.objects.filter(pk=device.pk))[device.id]
-            _, output = graph_ql_query(request, device, settings.sot_agg_query.query)
+            settings = get_device_to_settings_map(queryset=Device.objects.filter(pk=device.pk))
+            if device.id in settings:
+                _, output = graph_ql_query(request, device, settings[device.id].sot_agg_query.query)
+            else:
+                output = {"Error": f"{device.name} does not map to a Golden Config Setting."}
```

Comment:
I believe we either do not need this, or we should just raise an error.

Reply:
Done